firewall: live view improvments

devNan0 opened the pull request:
Add:
- `does not contain` filter condition
- `is not` filter condition
- `proto` filter

Changed:
- Sorting of `interface_name`

Remove:
- Duplicate `dst` entry in `filter_tag` list

Closes: https://github.com/opnsense/core/issues/4299 and https://github.com/opnsense/core/issues/4365 point 3

**Work in Progress**

lattera commented:
I could make use of this in several OPNsense deployments. Any objections to merging this?

AdSchellevis commented:
yes, it's in draft state for good reasons.

AdSchellevis commented:
@lattera while here, can you check https://github.com/HardenedBSD/hardenedBSD/issues/388 by the way. one of our people found the reason why dtrace didn't work on our end.

lattera commented:
> yes, it's in draft state for good reasons.

I'm curious what those reasons are.

devNan0 commented:
Ahhh... Sorry shame on me... forgot about the last points. Will commit and finish this draft in the next days.

Sorry for the delay :(

devNan0 commented:
Changed sorting of `interface_name`
Before:
![image](https://user-images.githubusercontent.com/49376203/102698764-23bf1e80-4240-11eb-8aae-0e8adee35634.png)
After:
![image](https://user-images.githubusercontent.com/49376203/102698774-3c2f3900-4240-11eb-8cc9-9106c0420cee.png)

Added `proto` filter option to solve https://github.com/opnsense/core/issues/4365 point 4

But i don´t really know how to implement `host` and `port` filter without a mess in the code.
@AdSchellevis maybe you could give me a hint.

TheLinuxGuy commented:
Thanks for working on this! I was also looking for "is not" filtering.

AdSchellevis commented:
@devNan0 thanks, I've pulled it in in different commits while testing the functionality. not sure what the intent of "proto" was since protoname is already the translated variant (udp, tcp). feel free open another ticket / PR to discuss further additions.
